src/js/angular/aclmanagement/controllers.js

```diff
@@ -193,6 +193,10 @@ function AclManagementCtrl($scope, $location, toastr, AclManagementRestService,
      * @param {string} scope
      */
     $scope.saveRule = (scope) => {
+        if (!$scope.rulesModel.isRoleValidLength($scope.editedRuleScope, $scope.editedRuleIndex)) {
+            notifyInvalidRoleLength();
```

Comment:
I don't like these toastrs but as I can see, the duplicated rule validation is implemented the same way, although this one is different because it validates single field and the other one validates the whole rule. Maybe the validation in this case should be done differently. Probably some error status should be added on the role field and the save button should be disabled until valid role is provided by the user.

Reply:
Alright, I'll do it as suggested